### PR TITLE
Scripts to build a binary distribution

### PR DESCRIPTION
**Pull request by christiaanb**
* Based on Nix. Requires Nix on the host, but _not_ on the target.

* bindist will contain
  * `/nix`: Nix-derivation of:
      1. bash
      2. ghc with clash-prelude and clash-lib
      3. clash
  * `/bin`: symlinks for the above executables
  * `nix-user-chroot`: fakechroot to map `./nix` to `/nix`
  * `clash`: shell-wrapper using fakechroot to `clash`
  * `clashi`: shell-wrapper using fakechroot to `clashi`

Simply run `./bindist/linux/mkBinDist.sh` to create a `./clash-bindist.tar.xz` archive.

Unpack said archive anywhere using:

`tar xf clash-bindist.tar.xz --one-top-level`

Then you can simply run:

`./clash-bindist/clash` or
`./clash-bindist/clashi`

or add `./clash-bindist` to you path.

### CODE REVIEW

File: bindist/linux/mkBinDist.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+# create symlinks inside target's usr/local/bin to every executable in the derivation (non-transitively)
+mkdir -p bin
+for binary in $(find ${drv}/bin); do
+  ln -s ${binary} bin/$(basename ${binary})
+done
+
+# finally, pack the full closure of the derivation with the symlinks
+# we just created
+tar cJf $tarball --owner 0 --group 0 bin clash clashi nix-user-chroot $(nix-store -qR $drv)
```
Comment (martijnbastiaan) [inline bindist/linux/mkBinDist.sh]: Maybe

```bash
tar -I pxz -cf $tarball [..]
```

For >1 thread.

File: bindist/linux/mkBinDist.sh [hunk 2]
```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env bash
+set -e
+
+# Path to thhe nix file containing the derivation we want to package
```
Comment (TMVector) [inline bindist/linux/mkBinDist.sh]:
```suggestion
# Path to the nix file containing the derivation we want to package
```

Comment (christiaanb) [reply bindist/linux/mkBinDist.sh]: Fixed

File: bindist/linux/mkBinDist.sh (hunk 1 above)
Comment (christiaanb) [reply bindist/linux/mkBinDist.sh]: Fixed by other means

**Pull request merged**